src/app/components/MainButton/MainButton.tsx

```diff
@@ -0,0 +1,25 @@
+import type { ReactNode } from 'react';
+import React from 'react';
+import styled from 'styled-components';
+
+type ButtonProps = {
+  children: ReactNode;
+  handleClick: () => void;
+};
+export default function MainButton({
+  children,
+  handleClick,
+}: ButtonProps): JSX.Element {
+  return <Button onClick={handleClick}>{children}</Button>;
+}
```

Comment:
This doesn't need to be another component returning a styled component, you could just export the styled component and pass it your ButtonProps type :)